.github/workflows/run-kola-tests.yaml

```diff
@@ -0,0 +1,137 @@
+name: "Run kola tests"
+on:
+  workflow_call:
+    inputs:
+      arch:
+        type: string
+        default: '["amd64", "arm64"]'
+        description: |
+          The architectures to run tests for. This is an array of architectures
+          that can be deserialized in JSON and results in an array of strings.
+        required: true
+      workflow_name_or_id:
+        type: string
+        required: true
+        default: ci.yaml
+        description: |
+          The workflow ID from where we'll download the artifacts to be tested.
+      workflow_run_id:
+        type: string
+        required: true
+        description: |
+          The run ID of the workflow specified in workflow_name_or_id
+      parallel_tests:
+        type: string
+        required: true
+        default: "2"
+        description: |
+          The number of parallel tests to run.
+
+jobs:
+  tests:
+    name: "Build Flatcar packages"
+    runs-on:
+      - self-hosted
+      - debian
+      - kola
+      - ${{ matrix.arch }}
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: ${{fromJSON(github.event.inputs.arch)}}
+
+    steps:
+      - name: Prepare machine
+        shell: bash
+        working-directory: ${{ github.workspace }}
+        run: |
+          sudo rm /bin/sh
+          sudo ln -s /bin/bash /bin/sh
+          sudo apt-get install -y ca-certificates curl gnupg lsb-release qemu-system qemu-user-static git bzip2 jq dnsmasq
+          sudo systemctl stop dnsmasq
+          sudo systemctl mask dnsmasq
+
+          # Install Docker-CE
+          sudo mkdir -p /etc/apt/keyrings
+          curl -fsSL https://download.docker.com/linux/debian/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+          echo \
+            "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian \
+            $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+          sudo apt-get update
+          sudo apt-get install -y docker-ce docker-ce-cli containerd.io docker-compose-plugin
+
+          # Install latest Go
```

Comment:
I think we can skip the Go installation because we're using Kola from the Docker version, so no need to build it from sources. :)